backend/src/main/java/fk/prof/backend/model/request/RecordedProfileHeaderParser.java

```diff
@@ -0,0 +1,106 @@
+package fk.prof.backend.model.request;
+
+import com.google.protobuf.CodedInputStream;
+import fk.prof.backend.exception.HttpFailure;
+import io.vertx.core.buffer.Buffer;
+import recording.Recorder;
+
+import java.io.IOException;
+import java.util.zip.Adler32;
+import java.util.zip.Checksum;
+
+public class RecordedProfileHeaderParser {
+  private Integer encodedVersion = null;
+  private Integer headerLength = null;
+  private Recorder.RecordingHeader recordingHeader = null;
+  private Integer checksumValue = null;
+
+  private Checksum headerChecksum = new Adler32();
+  private boolean headerParsed = false;
+  private int maxAllowedBytesForRecordingHeader;
+
+  public RecordedProfileHeaderParser(int maxAllowedBytesForRecordingHeader) {
+    this.maxAllowedBytesForRecordingHeader = maxAllowedBytesForRecordingHeader;
+  }
+
+  /**
+   * Returns true if all fields of header have been read and checksum validated, false otherwise
+   *
+   * @return returns if header has been parsed or not
+   */
+  public boolean isParsed() {
+    return this.headerParsed;
+  }
+
+  /**
+   * Returns {@link RecordedProfileHeader} if {@link #isParsed()} is true, null otherwise
+   *
+   * @return
+   */
+  public RecordedProfileHeader get() {
+    if (!this.headerParsed) {
+      return null;
+    }
+    return new RecordedProfileHeader(this.encodedVersion, this.recordingHeader);
+  }
+
+  /**
+   * Reads buffer and updates internal state with parsed fields. Returns the starting unread position in outputstream
+   *
+   * @param codedInputStream
+   * @return starting unread position in buffer
+   */
+  public int parse(CodedInputStream codedInputStream, Buffer underlyingBuffer, int currentPos) throws HttpFailure {
+    try {
+      if (!headerParsed) {
```

Comment:
Not really right.
1. Header has encoding version, length, proto message, checksum over (encoding version, length, proto message).
Whereas, wse has length, proto message, checksum over (length, proto message)
2. I am using proto message's instance method (parseFrom) which is specific to a proto message object and there is no generic interface in protobuf which implements this method. Don't see how I can abstract out the usage of that method
3. Other than recorder header and wse, there is no other place where we are doing this. There is case to be made where we will be doing something similar in userapi for aggregated profile, but over there Gaurav was not using var encoding for persisting uint32 fields and parsing logic will have to change anyways.

Can discuss this offline if there is a cleaner way to abstract this out. I thought about this but let it be when I was coding it